File: 01-getting-started/src/scripts/syntax.js
// define attributes / variables
//   number
//   string
//   boolean
//   array
//   dictionary / objects
//   undefined
// sample if / else
// functions
//   parameters
//   returns
// arrays
//   add to the front
//   add to the end
//   update values
// loops 
//   for
//   for/in
//   while
//   do while
//   forEach (with array and function)
// Objects / Dictionaries
//   declare object
//   lookup key to retrieve value


const functionGroup1 = {
    // Function below determines the type of varibles;
    defineVar: (item) => {
        if (typeof item === "number") return "number";
        if (typeof item === "string") return "string";
        if (typeof item === "boolean") return "boolean";
        if (typeof item === "object"){
            if (Array.isArray(item) === true) return "array";
            if (item === null)  return "null";
            return "object";
        }
        if (typeof item === "undefined") return "undefined";
    },

    // Function below test if else conditions;
    isOverFifty: (item) => {
        if (item > 49) return "success";
        else return "failed";
    },

    // Function below shows how parameter and return works in a function;
    defineFunction: (para1, para2) => {
        return para1 + para2;
        // function exits after it returns
    },

    // Function below excutes several array methods;
    addArrStart: (arr, item) => {
        arr.unshift(item);
        let newArr = arr;
        return newArr;
    },
    addArrEnd: (arr, item) => {
        arr.push(item);
        let newArr = arr;
        return newArr;
    },
    changeArrIdx: (arr, item, idx) => {
        arr[idx] = item;
        let newArr = arr;
        return newArr;
    },

    // Functions below runs different loop method
    forLoop: (arr) => {
        //combine string in an arr
        let myString = '';
        for (let i = 0; i<arr.length; i++){
            myString += arr[i]
        };
        return myString;
    },
    forInLoop: (obj) => {
        //combine string in an obj
        let myString2 = '';
        let item;
        for (item in obj){
            myString2 += item
        };
        return myString2;
    },
    whileLoop: (data) => {
        //adding string if condition is less than three
        let i = 0;
        let myString3 = '';
        while (i < 3){
            myString3 += data;
            i++;
        };
        return myString3;
    },
    doWhileLoop: (data) => {
        //combine string once
        let i = 0;
        let myString4 = '';
        do {
            myString4 += `${data} is received`;
            i++;
        }
        while (i < 0);
        return myString4;
    },
    forEachfunction: (arr) => {
        //combine string in an arr
        let colors ='the three colors are';
        arr.forEach((x)=> colors+= ` ${x}`);
        return colors;
    },

    //object manipulation
    objReadFirstKey: (obj) => {
        //read first key/value of the object
        let arr = Object.entries(obj);
        return `the first property is ${arr[0][0]} and the value is ${arr[0][1]}`
    }
};

export default functionGroup1; 

File: 01-getting-started/syntax.py
# define attributes / variables
# number (int float in python)
# string
# boolean
# array
# dictionary / objects
# None
# if / else
# functions
# parameters
# returns
# arrays
# add to the front
# add to the end
# update values
# loops
# for/in
# Objects / Dictionaries
# declare object
# lookup key to retrieve the value


def define_type(item):
    if type(item) is int:
        return 'integer'
    elif type(item) is float:
        return 'float'
    elif type(item) is str:
        return 'string'
    elif type(item) is bool:
        return 'boolean'
    elif type(item) is dict:
        return 'dictionary'
    elif type(item) is list:
        return 'list'
    elif type(item) is tuple:
        return 'tuple'
    elif type(item) is set:
        return 'set'
    elif type(item) is None:
        return None


def is_over_fifty(num):
    if num > 50:
        return 'success'
    else:
        return 'failed'

# list methods


def my_function(num1, num2):
    return num1 + num2


def list_add_start(ls, item):
    ls.insert(0, item)
    return ls


def list_add_end(ls, item):
    ls.append(item)
    return ls


def list_update_value(ls, item, idx):
    ls[idx] = item
    return ls

# loop methods


def for_in_loop(str):
    new_list = []
    for letter in str:
        new_list.append(letter)
    return new_list


def while_loop(data):
    i = 0
    my_string = ''
    while(i < 3):
        my_string += data
        i += 1
    return my_string

def dict_read_first_key(obj):
    my_list = list(obj.items())
    return f'the first property is {my_list[0][0]} and the value is {my_list[0][1]}'

def zip_my_lists(list1, list2):
    new_list = list(zip(list1, list2))
    return new_list

print(zip_my_lists([1,2,3],[4,5,6]))

def merge_sets(set1, set2):
    new_set = set1.union(set2)
    return new_set

print(merge_sets({1,2},{4,5}))

def email(fname, lname):
    return f'{fname.lower()}.{lname.lower()}@evolveu.ca'